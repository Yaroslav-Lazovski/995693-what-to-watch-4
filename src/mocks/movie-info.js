export const movieInfo = {
  title: `Oblivion`,
  poster: `img/oblivion.jpg`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};
