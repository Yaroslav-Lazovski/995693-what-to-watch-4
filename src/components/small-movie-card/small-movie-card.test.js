import React from "react";
import renderer from "react-test-renderer";
import {Router} from "react-router-dom";

import history from "../../history";
import SmallMovieCard from "./small-movie-card.jsx";

import {movieInfo} from "../../mocks/movie-info.js";


describe(`SmallMovieCard Snapshot`, () => {
  it(`Should small movie card render correctly`, () => {
    const tree = renderer
      .create(
          <Router
            history={history}
          >
            <SmallMovieCard
              movie={movieInfo}
              isPlaying={false}
              onMouseEnter={() => {}}
              onMouseLeave={() => {}}
              onStartPlaying={() => {}}
              onStopPlaying={() => {}}
            />
          </Router>, {
            createNodeMock: () => {
              return {};
            }
          }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
