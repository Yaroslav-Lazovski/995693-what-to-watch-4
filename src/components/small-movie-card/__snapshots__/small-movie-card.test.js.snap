// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallMovieCard Snapshot Should small movie card render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Oblivion"
      height="175"
      src="img/oblivion.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Oblivion
    </a>
  </h3>
</article>
`;
